src/miniwdl_slurm/__init__.py

```diff
@@ -94,10 +110,19 @@ def _slurm_invocation(self):
             # Round to the nearest megabyte.
             srun_args.extend(["--mem", f"{round(memory / (1024 ** 2))}M"])
 
+        gpu = self.runtime_values.get("gpu", None)
+        if gpu:
+            num_gpu = self.runtime_values.get("num_gpu", 1)
+            srun_args.extend(["--gres", f"gpu:{num_gpu}"])
```

Comment:
please use `--gpus-per-task` as a flag instead and also change the `num_gpu` variable to `gpus_per_task`. That will be more transparent to the slurm user.

I did a check and see that `gpu` is a keyword that is already in the WDL SPEC. I couldn't find any word for number of gpus. Should we use `gpus_per_task` here or `slurm_gpus_per_task`? @DavyCats  what do you think?

Reply:
I don't think the `--gpus-per-task` works on our cluster. I used the `--gres` when I needed GPUs.

As for the runtime attribute name, I guess `gpus_per_task` is fine, since the number of GPUs to use isn't slurm specific.

Reply:
Ah. but `task` is a bit specific to SLURM and there is no current OpenWDL spec keyword. Let's choose a reasonably good one that might be used in the spec too. 
Let's go with `number_of_gpus` which defaults to 1 if `gpu` is set. 
@williamrowell can you change `num_gpu` to `number_of_gpus` ? Then I will merge the PR and release so you can use it.

Reply:
> I don't think the `--gpus-per-task` works on our cluster. I used the `--gres` when I needed GPUs.
> 
> As for the runtime attribute name, I guess `gpus_per_task` is fine, since the number of GPUs to use isn't slurm specific.

I prefer `--gpus-per-task` as well, but I've encountered similar issues, and it seems like `--gres` works more consistently.  Alternatively, we could use a string attribute instead of an int to set the number of GPUs, like `"slurm-gpu-param": "--gres=gpu:2"`.

Reply:
> gpuCount is a known key for the [miniwdl linter](https://github.com/chanzuckerberg/miniwdl/blob/df1a122363562d64cce24ff9488e1356e42ddcb6/WDL/Lint.py#L1085-L1087). Would you rather use that instead, or does it indicate something different?

That would be excellent. Let's stay as close to miniwdl as possible.

> I prefer sbatch --gpus-per-task as well, but I've encountered similar issues, and it seems like sbatch --gres works more consistently. Alternatively, we could use a string attribute instead of an int to set the number of GPUs, like "slurm-gpu-param": "--gres=gpu:2".

I prefer "gpuCount". That is easier for users. If there are problems, people can report them here and we can fix them. Let's not have our decisions influenced by problems that have not occurred yet.

Reply:
I renamed `num_gpu` to `gpuCount` and tested again locally.  Given that this might be a pain to test, I attached my config, hello.wdl, and the run directory.

[miniwdl-slurm-test.tgz](https://github.com/miniwdl-ext/miniwdl-slurm/files/11144646/miniwdl-slurm-test.tgz)